bin/desi_archive_tilenight

```diff
@@ -117,33 +122,58 @@ else:
     reduxdir = args.prod
 
 if args.specstatus is None:
-    args.specstatus = os.path.expandvars('$DESI_ROOT/survey/ops/surveyops/trunk/ops/tiles-specstatus.ecsv')
+    if os.path.exists('./tiles-specstatus.ecsv'):
+        args.specstatus = os.path.abspath('./tiles-specstatus.ecsv')
+    else:
+        args.specstatus = os.path.expandvars(
+            '$DESI_ROOT/survey/ops/surveyops/trunk/ops/tiles-specstatus.ecsv')
 
 if not os.path.exists(args.specstatus):
     log.critical(f'Missing {args.specstatus}')
     sys.exit(1)
 
 log.info(f'Reading tiles from {args.specstatus}')
+if not args.dry_run and not os.access(args.specstatus, os.W_OK):
+    filename = os.path.basename(tiles.specstatus)
+    log.critical('Need write access to {filename} to update ZDONE and ARCHIVEDATE')
+    sys.exit(1)
+
+log.info(f'Archiving tiles in {reduxdir}')
+os.chdir(reduxdir)
+
 tiles = Table.read(args.specstatus)
+log.info('Archiving tiles with SURVEY=main|sv3, ZDONE=false, QA=good')
 keep = (tiles['SURVEY'] == 'main') | (tiles['SURVEY'] == 'sv3')
-keep &= (tiles['ZDONE'] == 'true') & (tiles['ARCHIVEDATE'] > 0)
+keep &= (tiles['ZDONE'] == 'false') & (tiles['QA'] == 'good')
+if args.thru is not None:
+    log.info(f'Filtering to LASTNIGHT<={args.thru}')
+    keep &= tiles['LASTNIGHT'] <= args.thru
+if args.since is not None:
+    log.info(f'Filtering to LASTNIGHT>={args.since}')
+    keep &= tiles['LASTNIGHT'] >= args.since
 if args.night is not None:
     log.info(f'Filtering to just LASTNIGHT={args.night}')
     keep &= tiles['LASTNIGHT'] == args.night
-if args.tileid is not None:
-    log.info(f'Filtering to just TILEID={args.tileid}')
-    keep &= tiles['TILEID'] == args.tileid
+if args.tileids is not None:
+    tileids = [int(t) for t in args.tileids.split(',')]
+    log.info(f'Filtering to just TILEIDs {tileids}')
+    keep &= np.isin(tiles['TILEID'], tileids)
+
+archivetiles = tiles[keep]
 
-tiles = tiles[keep]
+archivedate = datetime.datetime.now().strftime('%Y%m%d')
+ntiles = len(archivetiles)
+if ntiles == 0:
+    log.warning('No tiles to archive; exiting')
+    sys.exit(0)
 
-ntiles = len(tiles)
-log.info(f'{ntiles} tiles to archive')
+log.info(f'Archiving {ntiles} tiles using ARCHIVEDATE={archivedate}')
 
 #- find and read exposures table for that specprod
 specprod = os.path.basename(reduxdir)
 for expfile in [
-    f'{reduxdir}/tsnr-exposures.fits',
     f'{reduxdir}/exposures-{specprod}.fits',
```

Comment:
Should this also me made consistent with relative vs absolute paths?

Reply:
I hadn't changed this to local dirs because it didn't impact logging, but if you think that is more maintainable to be completely consistent with relative vs. absolute paths I could update it.

FWIW these lines changed because I wanted to check the new standard exposures filename first instead of the old name first.

Reply:
I misinterpreted the change to relative as operating in the local directory vs a predefined one. I now see that is not the intention here, so I'm fine with this staying as is.

And +1 for prioritizing the new name convention.